drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb

```diff
@@ -0,0 +1,86 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaYyaFollowupReport
+  class Report
+    include ArelHelper
+    include Filter::FilterScopes
+
+    def initialize(filter)
+      @filter = filter
+      @end_date = @filter.on
+      @start_date = @end_date - 2.months - 3.weeks
+      @late_date = @end_date - 3.months
+    end
+
+    attr_accessor :start_date, :late_date
+
+    def clients
+      client_scope.pluck(*columns.values).
+        map do |row|
+          Hash[columns.keys.zip(row)]
+        end.
+        sort_by { |row| row[:last_seen] }
+    end
+
+    def columns
+      window = ::Arel::Nodes::Window.new.partition(c_t[:id])
+      {
+        id: :id,
+        first_name: :FirstName,
+        last_name: :LastName,
+        engagement_date: she_t[:first_date_in_program].minimum.over(window),
+        last_seen: cls_t[:InformationDate].maximum.over(window),
+      }
```

Comment:
a new toy?

Reply:
Yep -- Rails 6 support for window functions... Line 30 is a little ugly but it produces `MAX("CurrentLivingSituation"."InformationDate") OVER (PARTITION BY "Client"."id")` which is nice!